drivers/interrupt_controller/intc_plic.c

```diff
@@ -34,8 +34,10 @@
  */
 #define PLIC_REG_PRIO_OFFSET 0x0
 #define PLIC_REG_IRQ_EN_OFFSET 0x2000
+#define PLIC_REG_IRQ_EN_SIZE 0x80
 #define PLIC_REG_REGS_OFFSET 0x200000
-#define PLIC_REG_REGS_THRES_PRIORITY_OFFSET 0
+#define PLIC_REG_REGS_SIZE 0x1000
+#define PLIC_REG_REGS_THRES_PRIORITY_OFFSET 0x1000
```

Comment:
```suggestion
#define PLIC_REG_REGS_THRES_PRIORITY_OFFSET 0
```
This was supposed to be the offset from `PLIC_REG_REGS_OFFSET + (n * PLIC_REG_REGS_SIZE)`, it should remains as `0`

Reply:
hey @ycsin, thanks for taking the time to review, really appreciated!
from the PLIC Specification:
```
...
base + 0x1FFFFC: Reserved
base + 0x200000: Priority threshold for context 0
base + 0x200004: Claim/complete for context 0
base + 0x200008: Reserved
...
base + 0x200FFC: Reserved
base + 0x201000: Priority threshold for context 1
base + 0x201004: Claim/complete for context 1
```
The threshold priority offset is in fact 0x1000. If we are defining a MACRO for the value, I think we should use the right value.

Reply:
The PLIC spec implies that they belongs to the same register alignment:

```
The base address of Priority Thresholds register block is located at 4K alignment starts from offset 0x200000.
```

```
The Interrupt Claim Process register is context based and is located at (4K alignment + 4) starts from offset 0x200000.
```

You might want to update `PLIC_REG_REGS_CLAIM_COMPLETE_OFFSET` as well to be consistent with the current change, i.e.:

```c
#define PLIC_REG_REGS_CLAIM_COMPLETE_OFFSET (PLIC_REG_REGS_SIZE + sizeof(uint32_t))

/* or */

#define PLIC_REG_REGS_CLAIM_COMPLETE_OFFSET (PLIC_REG_REGS_SIZE + 4)

/* or */

#define PLIC_REG_REGS_CLAIM_COMPLETE_OFFSET 0x1004
```

I'd rather follow the [Linux's plic driver](https://github.com/torvalds/linux/blob/3b47bc037bd44f142ac09848e8d3ecccc726be99/drivers/irqchip/irq-sifive-plic.c#L56-L59) convention though:

```c
/*
 * Each hart context has a set of control registers associated with it.  Right
 * now there's only two: a source priority threshold over which the hart will
 * take an interrupt, and a register to claim interrupts.
 */
#define CONTEXT_BASE			0x200000
#define     CONTEXT_SIZE		0x1000
#define     CONTEXT_THRESHOLD		0x00
#define     CONTEXT_CLAIM		0x04
```

Reply:
@ycsin, I have aligned with Linux's convention

Reply:
The end result is the same, but you should fix it in the first commit instead

Reply:
@ycsin done